xm-commons-logging-trace/src/main/java/com/icthh/xm/commons/logging/trace/TraceService.java

```diff
@@ -0,0 +1,31 @@
+package com.icthh.xm.commons.logging.trace;
+
+import brave.spring.web.TracingClientHttpRequestInterceptor;
+import com.icthh.xm.commons.lep.spring.ApplicationLepProcessingEvent;
+import com.icthh.xm.lep.api.ContextScopes;
+import com.icthh.xm.lep.api.LepManager;
+import com.icthh.xm.lep.api.LepProcessingEvent;
+import com.icthh.xm.lep.api.ScopedContext;
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.ApplicationListener;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class TraceService implements ApplicationListener<ApplicationLepProcessingEvent> {
+
+    private final LepManager lepManager;
+    private final TracingClientHttpRequestInterceptor tracingClientHttpRequestInterceptor;
+
+    @Override
+    public void onApplicationEvent(ApplicationLepProcessingEvent event) {
+        if (event.getLepProcessingEvent() instanceof LepProcessingEvent.BeforeExecutionEvent) {
+            ScopedContext context = this.lepManager.getContext(ContextScopes.EXECUTION);
+            context.setValue("traceService", this);
+        }
+    }
+
+    public TracingClientHttpRequestInterceptor getTraceInterceptor() {
```

Comment:
how about do not expose TracingClientHttpRequestInterceptor, and just delegate method that you need, to have ability to change implementation

Reply:
we need all TracingClientHttpRequestInterceptor class, because we pass him in lep when create new RestTemplate

`def restTemplate = new RestTemplate()
ClientHttpRequestInterceptor sleuthInterceptor = lepContext.traceService?.getTraceInterceptor()
List<ClientHttpRequestInterceptor> interceptors =
                new ArrayList<ClientHttpRequestInterceptor>(restTemplate.getInterceptors())
interceptors.add(0, sleuthInterceptor)
restTemplate.setInterceptors(interceptors)`